### PR TITLE
addressed reviews for one resource and datasource

### PR DESCRIPTION
**Pull request by AnikaAgiwal2711**
# Description
A few sentences describing the overall goals of the pull request's commits.

# GitHub Issues
List the GitHub issues impacted by this PR:

| GitHub Issue # |
| -------------- |
| |

# Checklist:

- [ ] I have performed a self-review of my own code to ensure there are no formatting, vetting, linting, or security issues
- [ ] I have verified that new and existing unit tests pass locally with my changes
- [ ] I have not allowed coverage numbers to degenerate
- [ ] I have maintained at least 80% code coverage
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Backward compatibility is not broken

# How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Please also list any relevant details for your test configuration

- [ ] Test A
- [ ] Test B

### CODE REVIEW

File: docs/resources/storage_pool.md [hunk 1]
```diff
@@ -61,38 +61,58 @@ limitations under the License.
...
+  rebuild_rebalance_parallelism = 5
+  fragmentation                 = false
 }
 
 output "created_storagepool" {
   value = powerflex_storage_pool.sp
 }
 ```
 
+After the execution of above resource block, storage pool would have been created on the PowerFlex array. For more information, Please check the state file.
```
Comment (gokul-srivathsan) [inline docs/resources/storage_pool.md:114]: Existing text: "Please check the state file" 
Proposed text: "Please check the terraform state file"

File: docs/data-sources/storage_pool.md [hunk 2]
```diff
@@ -19,12 +19,12 @@ linkTitle: "powerflex_storage_pool"
...
-  This data-source can be used to fetch information related to the storage pools from a PowerFlex array.
+  This datasource is used to query the existing storage pools from PowerFlex array. The information fetched from this datasource can be used for getting the details / for further processing in resource block.
 ---
 
 # powerflex_storage_pool (Data Source)
 
-This data-source can be used to fetch information related to the storage pools from a PowerFlex array.
+This datasource is used to query the existing storage pools from PowerFlex array. The information fetched from this datasource can be used for getting the details / for further processing in resource block.
 
 ~> **Note:** Exactly one of `protection_domain_name` and `protection_domain_id` is required.
```
Comment (gokul-srivathsan) [inline docs/data-sources/storage_pool.md:29]: Proposed Text: 
Every input config file must have either a protection_domain_name or protection_domain_id field.Not both.
Every input config file must have either storage_pool_names or storage_pool_ids attribute can be given at a time, not both.

Comment (AnikaAgiwal2711) [reply docs/data-sources/storage_pool.md:29]: As discussed with Gokul, We need to change "and" to "or" . Also at the end of the sentence we'll be adding "Not both can be provided"

File: powerflex/provider/mdm_cluster_resource.go [hunk 3]
```diff
@@ -67,8 +67,8 @@ func (d *mdmClusterResource) Schema(_ context.Context, _ resource.SchemaRequest,
 
 // MdmClusterResourceSchema defines the schema for Mdm resource
 var MdmClusterResourceSchema schema.Schema = schema.Schema{
-	Description:         "This resource can be used to manage MDM cluster on a PowerFlex array.",
-	MarkdownDescription: "This resource can be used to manage MDM cluster on a PowerFlex array.",
+	Description:         "This resource can be used to manage MDM cluster on a PowerFlex array. Supports adding or removing standby MDMs, migrate to/from a 3-node or 5-node cluster, changing MDM ownership, changing performance profile, and renaming MDMs.",
```
Comment (AnikaAgiwal2711) [inline powerflex/provider/mdm_cluster_resource.go:70]: Can we write this: " migrate to/from a 3-node or 5-node cluster " as :  _**"migrate from 3-node to 5-node cluster or vice -versa"**_

**Comment by AnikaAgiwal2711** [comment]
Also , Is there any more information which you want to mention on the examples file of mdm

File: powerflex/provider/mdm_cluster_resource.go (hunk 3 above)
Comment (AnikaAgiwal2711) [reply powerflex/provider/mdm_cluster_resource.go:70]: We also need to mention about Import that we're not supporting the import for mdm.

File: examples/resources/powerflex_cluster/resource.tf [hunk 4]
```diff
@@ -20,67 +20,101 @@ limitations under the License.
...
+  # Security Related Field
+  mdm_password                            = "Password"
+  lia_password                            = "Password"
+
+  # Advance Security Configuration
+  allow_non_secure_communication_with_lia = false
+  allow_non_secure_communication_with_mdm = false
+  disable_non_mgmt_components_auth        = false
+
+  # Cluster ronfiguration related fields
```
Comment (AnikaAgiwal2711) [inline examples/resources/powerflex_cluster/resource.tf:33]: Configuration typo

File: examples/resources/powerflex_cluster/resource.tf [hunk 5]
```diff
@@ -20,67 +20,101 @@ limitations under the License.
...
+    {
+      # MDM Configuration Fields 
+      ips                     = "10.10.10.1",
+      username                = "root",
+      password                = "Password",
+      operating_system        = "linux",
+      is_mdm_or_tb            = "primary",
+      mdm_ips                 = "10.10.10.1",
+      mdm_mgmt_ip                 = "10.10.10.1",
+      mdm_name                 = "MDM_1",
```
Comment (AnikaAgiwal2711) [inline examples/resources/powerflex_cluster/resource.tf:44]: Formatting issue

File: examples/resources/powerflex_cluster/resource.tf [hunk 6]
```diff
@@ -20,67 +20,101 @@ limitations under the License.
...
+      is_mdm_or_tb         = "TB",
+      is_sds               = "No",
+      is_sdc               = "yes",
+      sdc_name             = "sdc3",
+      perf_profile_for_sdc = "compact",
+      is_rfcache           = "No",
+      is_sdr               = "No",
+    },
+  ]
+  storage_pools = [
```
Comment (AnikaAgiwal2711) [inline examples/resources/powerflex_cluster/resource.tf:112]: Storage Pool configuration Fields

File: docs/resources/cluster.md [hunk 7]
```diff
@@ -19,24 +19,24 @@ linkTitle: "powerflex_cluster"
...
 subcategory: ""
 description: |-
-  This resource can be used to install the PowerFlex Cluster.
+  This terraform resource is used to deploy PowerFlex Cluster. We can Create an Delete the PowerFlex Cluster using this resource. We can also Import an existing Cluster of PowerFlex.
 ---
 
 # powerflex_cluster (Resource)
 
-This resource can be used to install the PowerFlex Cluster.
+This terraform resource is used to deploy PowerFlex Cluster. We can Create an Delete the PowerFlex Cluster using this resource. We can also Import an existing Cluster of PowerFlex.
```
Comment (AnikaAgiwal2711) [inline docs/resources/cluster.md:27]: We can Create **_and_** Delete the

**Review by gokul-srivathsan — APPROVED** [review]
LGTM

**Review by Krunal-Thakkar — APPROVED** [review]

**Comment by AnikaAgiwal2711** [comment]
run e2e test

**Comment by AnikaAgiwal2711** [comment]
run e2e test

**Comment by AnikaAgiwal2711** [comment]
run e2e test

**Comment by AnikaAgiwal2711** [comment]
run e2e test

**Pull request merged**